Name: Pandora Ad Fix
Version: 1.3.5-1
Author: VincentLaw
Description: Fixes empty advertisement box issue. This patch is not intended to disable ads entirely, and never will! It will only keep Pandora from displaying the ad box in empty cases.

--- .orig/media/cryptofs/apps/usr/palm/applications/com.palm.pandora/app/core/ad_service.js
+++ /media/cryptofs/apps/usr/palm/applications/com.palm.pandora/app/core/ad_service.js
@@ -36,7 +36,7 @@
 		var txt = response.responseText;
 		Util.log("ad text: " + txt);
 		var adData = this.parseAdJSON(txt);
-		if (adData != null && adData.imageUrl != null) {
+		if (adData != null && adData.imageUrl != null && adData.imageUrl != "") {
 			MessageDispatcher.dispatchMessage(AppConstants.EVENT_ON_DISPLAY_NOW_PLAYING_AD, adData);
 		}
 	},
\ No newline at end of file
